pages/api/plants/[id].js

```diff
@@ -0,0 +1,40 @@
+import dbConnect from "@/db/connect";
+import Plant from "@/db/models/PlantsSchema";
+
+export default async function handler(request, response) {
+  const { id } = request.query;
+  await dbConnect();
+
+  switch (request.method) {
+    case "GET":
+      const plant = await Plant.findById(id).populate("isOwned");
```

Comment:
Why populate?
I recommend to add the isOwned property=false as a default to every dataset in the DB.